stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientState.kt

```diff
@@ -0,0 +1,70 @@
+package io.getstream.chat.android.client.clientstate
+
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.utils.State
+
+internal sealed class ClientState : State {
+    object Idle : ClientState()
+    sealed class AnonymousUserState : ClientState() {
+        sealed class AnonymousUserPending : AnonymousUserState() {
+            object AnonymousPendingWithoutToken : AnonymousUserPending()
+            class AnonymousPendingWithToken(val token: String) : AnonymousUserPending()
```

Comment:
It would simplify readability a lot if we used shorter names for all these states. For example, right now one of the inner ones here is called `ClientState.AnonymousUserState.AnonymousUserPending.AnonymousPendingWithToken`. This repeats a lot of information in the naming of the enclosing state. Could just be `ClientState.Anonymous.Pending.WithToken`, for example.

Reply:
Good idea. Did it